pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -317,12 +323,16 @@ func NewTargetProxy(tcpBindAddress string, tcpBindPort int, serverTLSConfig *tls
 
 }
 
+func isLoopbackAddress(ipAddress string) bool {
+	return (strings.Contains(ipAddress, "127.0.0.1") || strings.Contains(ipAddress, "[::1]"))
```

Comment:
I am trying to remove hard-coded addresses form the code,
I will appreciate if you will export those addresses to variables :)

Reply:
np, to put them as a var in that specific function is good enough?

Reply:
Thanks!
We can add those ports to this file [const block ](https://github.com/kubevirt/kubevirt/blob/85435dd0711fb4773885ccfad27f9627ed325279/pkg/virt-handler/migration-proxy/migration-proxy.go#L38) and then pass them that function

Reply:
@oshoval please explain why this change is needed while the other changes related to using ::1 instead of 127.0.0.1 were dropped from the PR?

Reply:
Alona
Imo the function should be robust to both derivates of ::1 
which are both the same thing in different notation.

Also during development i saw a case where i had to use it, i will double check now, if its still necessary because there were changes from that point of time.

Reply:
> the other changes are purely for supported futuristic configuration bindv6only

Are these facts or assumptions? How do you know if the needed works with IPv6 and not IPv4?
You need to explain us how it works so we will be convinced that this point of yours is correct.
`bind6only` is totally irrelevant here, that is an implementation of socket details and you have no idea how the application is actually listening to the address (they may be listening to one or two sockets).

Reply:
see please my first post
https://stackoverflow.com/questions/6343747/ipv6-socket-creation
bindv6only is not just socket wise, its kernel wise
cat /proc/sys/net/ipv6/bindv6only

this link was groomed with several people
that ::1 can be used as long as bindv6only=0 
this is what was done on ovirt according Alona

and we dont have a configuration to test it,
imo it isnt related to this PR because its two different things:
1. support migration on ipv6 (rest of the changes here beside this function of isloopback that i can remove and move to the other PR)
2. the other PR is the one that will support a mode (bindv6only) that we even dont support and can test right now.

Reply:
> see please my first post and this link there
> https://stackoverflow.com/questions/6343747/ipv6-socket-creation
> bindv6only is not just socket wise, its kernel wise
> cat /proc/sys/net/ipv6/bindv6only

That system wide parameter overwrites the ability for a socket to be opened as a dual socket. How the application opens the socket is not explaining me how all the dots connect in this flow.

> 
> this link was groomed with several people
> that ::1 can be used as long as bindv6only=0
> this is what was done on ovirt according Alona
> (use ::1 unless ipv6 is disabled, hence also case of bindv6only=1
> is supported, because we wont use 127.0.0.1 / 0.0.0.

`bind6only` is irrelevant IMO. The application may open sockets as a single stack or as dual stack, either way, it should work with IPv6.

> 
> and we dont have a configuration to test it,
> imo it isnt related to this PR because its two different things:
> 
>     1. support migration on ipv6 (rest of the changes here beside this function of isloopback that i can remove and move to the other PR)
> 
>     2. the other PR is the one that will support a mode (bindv6only) that we even dont support and can test right now.

I only care at the moment about the correct support of IPv6 in the migration flow.
The current change normalizes the ipv6 address for the listener socket and adds loopback detection in case it is an IPv6 loopback.
It does all this in the proxy listener and ignores all other code that uses IPv4 only addresses (like 127.0.0.1).
I am looking for an explanation why the proxy listener is the only one needing change.
I think you also mentioned that even without the loopback change it works, so the same questions pops up for that.

Reply:
the virt-handlers are the only place in migration that use IPv6
there are address of the pods that are used in the migration
and they are the ones that need the normalize [].

i can fail the migration and show that error directly,
i will enclose soon if needed.

indeed the loopback detection isnt needed, because we still use 127.0.0.1 in this PR
and as 127.0.0.1 is subset of IPv6 it is fine.

Is it fine that i will remove that func from this PR, and focus on the normalizing ?
The func will be mandatory in the 2nd PR (verified right now), and then i will handle all the comments of it.

Reply:
```
{"component":"virt-handler","level":"error","msg":"unable to create outbound leg of proxy to host fd00:10:244::37:39173","pos":"migration-proxy.go:392","reason":"dial tcp: address fd00:10:244::37:39173: too many colons in address","timestamp":"2020-05-10T12:42:02.271518Z"}
```
this is the error that line
https://github.com/kubevirt/kubevirt/pull/3392/files#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R220
fixed when handler of the source migration node tries to connect to the dst migration handler

Reply:
https://access.redhat.com/solutions/4291941
does this answer please ?